scipy/stats/tests/test_stats.py

```diff
@@ -6969,70 +6969,92 @@ def test_dist_perm(self):
-
-    # should return different value because this implementation solves
-    # odd behavior for large numbers found in original fortran code
+class TestPoissonETest:
+
+    @pytest.mark.parametrize("c1, n1, c2, n2, p_expect", (
+        # example from [1], 6. Illustrative examples: Example 1
+        [0, 100, 3, 100, 0.0884],
+        [2, 100, 6, 100, 0.1749]
+    ))
+    def test_paper_examples(self, c1, n1, c2, n2, p_expect):
+        res = stats.poisson_means_test(c1, n1, c2, n2)
+        assert_allclose(res.pvalue, p_expect, atol=1e-4)
+
+    @pytest.mark.parametrize("c1, n1, c2, n2, p_expect, alt, d", (
+        # These test cases are produced by the wrapped fortran code from the
+        # original authors. Using a slightly modified version of this fortran,
+        # found here, https://github.com/nolanbconaway/poisson-etest,
+        # I created additional tests.
+        [20, 10, 20, 10, 0.9999997568929630, 'two-sided', 0],
+        [10, 10, 10, 10, 0.9999998403241203, 'two-sided', 0],
+        [10, 10, 10, 10, 0.741441295832156, 'two-sided', .1],
+        [3, 100, 20, 300, 0.12202725450896404, 'two-sided', 0]
+    ))
+    def test_fortran_authors(self, c1, n1, c2, n2, p_expect, alt, d):
+        res = stats.poisson_means_test(c1, n1, c2, n2, alternative=alt, diff=d)
+        assert_allclose(res.pvalue, p_expect, atol=1e-6)
+
     def test_different_results(self):
+        # The implementation in Fortran is known to break down at higher
+        # counts and observations, so we expect different results.
         count1, count2 = 10000, 10000
         nobs1, nobs2 = 10000, 10000
-        _, pval = stats.poisson_means_test(count1, nobs1, count2, nobs2)
+        res = stats.poisson_means_test(count1, nobs1, count2, nobs2)
         with assert_raises(AssertionError):
-            assert_almost_equal(pval, 0.24866994128694545, decimal=5)
+            assert_allclose(res.pvalue, 0.248669941286945455)
 
     # the original fortran code by the author does not implement what
     # they said in the paper, this implementation fixes that
+
     def test_less_than_zero_lambda_hat2(self):
         count1, count2 = 0, 0
         nobs1, nobs2 = 1, 1
-        _, pval = stats.poisson_means_test(count1, nobs1, count2, nobs2)
+        res = stats.poisson_means_test(count1, nobs1, count2, nobs2)
         with assert_raises(AssertionError):
-            assert_almost_equal(pval, 0.0, decimal=1)
+            assert_almost_equal(res.pvalue, 0.0)
 
-    def test_non_int_args(self):
+    def test_non_int_args_error(self):
         count1, count2 = 0, 0
-        nobs1, nobs2 = 1, 0.7
-        alternative = 'less'
-        with assert_raises(TypeError):
-            stats.poisson_means_test(count1, nobs1, count2, nobs2,
-                                     alternative=alternative)
-
-    def test_negative_k(self):
-        count1, count2 = -1, 1
         nobs1, nobs2 = 1, 1
-        with assert_raises(ValueError):
```

Comment:
A lot of things raise errors of the same type, so it's important to test the message.